### PR TITLE
FFT: added comments

### LINKED ISSUE #144: Add haddock comments on all exposed functions/types.

### LINKED ISSUE #159: Clean up the json file

### PR DESCRIPTION
**Pull request by JTeeuwissen**
resolves #144
resolves #159

### CODE REVIEW

File: core/src/JsonLogic/IO/Operation.hs [hunk 1]
```diff
@@ -23,23 +23,25 @@ where
...
 import qualified Prelude as P
 
--- Default operators
+-- | A map of all the default operations.
 defaultOperations :: Operations
 defaultOperations = M.unions [arrayOperations, booleanOperations, dataOperations, miscOperations, numericOperations, stringOperations]
 
-arrayOperations, booleanOperations, dataOperations, numericOperations, stringOperations :: Operations
+-- | Groups of operations on similar data.
+arrayOperations, booleanOperations, dataOperations,  numericOperations, stringOperations :: Operations
```
Comment (musicismyalibi) [inline core/src/JsonLogic/IO/Operation.hs:37]: Why is there an extra space before `numericOperations`?

File: core/src/JsonLogic/Json.hs [hunk 2]
```diff
@@ -33,7 +41,7 @@ type Rule = Json
 -- | Data can be any kind of JSON value.
 type Data = Json
 
--- | An instance to show json in clear format for users
+-- An instance to show json in clear format for users
```
Comment (musicismyalibi) [inline core/src/JsonLogic/Json.hs:44]: I think the dot is missing here.

**Review by musicismyalibi — COMMENTED** [review]
I think JSON and json are not used consistently.

File: core/src/JsonLogic/Json.hs (hunk 2 above)
Comment (JTeeuwissen) [reply core/src/JsonLogic/Json.hs:44]:
```suggestion
-- An instance to show json in clear format for users.
```

File: core/src/JsonLogic/IO/Operation.hs (hunk 1 above)
Comment (JTeeuwissen) [reply core/src/JsonLogic/IO/Operation.hs:37]:
```suggestion
arrayOperations, booleanOperations, dataOperations, numericOperations, stringOperations :: Operations
```

**Comment by JTeeuwissen** [comment]
> I think JSON and json are not used consistently.

Please add change proposals

**Comment by musicismyalibi** [comment]
The pure and IO `Type.hs` files still contain some JSON instead of Json.

**Review by musicismyalibi — APPROVED** [review]

**Pull request merged**